src/Shape.cpp

```diff
@@ -10,6 +10,27 @@
 
 namespace geo {
 
+/**
+ * @brief Check whether a point p is within distance radius of the line segment. The starting point is described by @b a and the vector from the starting point to the end point @b ab
```

Comment:
One property of this function which is not readily apparant is that it will return true only if the projection of p on line ab lies within the line segment ab. 
The reason for this is that we have already checked the endpoints of the lines in the main function.